### PR TITLE
Modify past UC to fit the current app

### LINKED ISSUE #295: Add use case for unlabelCommand

### LINKED ISSUE #296: Add use cases for labelCommand

### LINKED ISSUE #297: Add use cases for redoCommand

### LINKED ISSUE #298: Add use cases for undoCommand

### PR DESCRIPTION
**Pull request by Ma-Yueran**
Add UC for new features (label, unlabel, undo, redo)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -312,262 +312,282 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-1. User requests to tag a certain file using the absolute file path.
-2. HelloFile recognises the file path to be valid.
-3. HelloFile creates the tag and display the fileInfo.
+1. User requests to tag a file with a tag name, the file path and a label.
+2. HelloFile creates the tag and informs the user.
 
     Use case ends.
 
 **Extensions**
```
Comment (anthony6401) [inline docs/DeveloperGuide.md:325]: Should we put that a label can be invalid as well in the extension?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -312,262 +312,282 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-* 2a. HelloFile detects the file path is invalid.
+* 2a. HelloFile detects the file path entered to be invalid.
 
-    * 2a1. HelloFile prompts the user that the file path is wrong.
+    * 2a1. HelloFile prompts the user that the file path entered is invalid.
 
         Use case resumes from step 1.
 
-* 2b. HelloFile detects no tag input.
+* 2b. HelloFile detects the tag entered to be invalid.
```
Comment (anthony6401) [inline docs/DeveloperGuide.md]: You mean tag name instead of tag?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Ma-Yueran) [reply docs/DeveloperGuide.md:325]: Ooo, ya, I will add that

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Ma-Yueran) [reply docs/DeveloperGuide.md]: yes, should I change tag to tag name?

Comment (Ma-Yueran) [reply docs/DeveloperGuide.md]: ok, I changed tag to tag name now

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -313,287 +313,289 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-* 2b. HelloFile detects no tag input.
+* 2b. HelloFile detects the file path entered to be invalid.
 
-    * 2b1. HelloFile prompts the user for a tag.
+    * 2b1. HelloFile prompts the user that the file path entered is invalid.
 
         Use case resumes from step 1.
+        
+* 2c. HelloFile detects the label entered to be invalid.
```
Comment (luo-git) [inline docs/DeveloperGuide.md]: to be invalid -> is invalid

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -313,287 +313,289 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-* 3b. HelloFile detects no default application to open the file.
+* 2c. HelloFile detects that there is no permission to open the file.
 
-	* 3b1. HelloFile prompts the user to choose an application to open the file.
+	* 2c1. HelloFile prompts the user that the app does not have the permission to open the file.
 
 	    Use case resumes form step 1.
 
-* 3c. HelloFile encounters an exception when opening the file.
```
Comment (luo-git) [inline docs/DeveloperGuide.md:387]: This should not be deleted?

**Review by luo-git — CHANGES_REQUESTED** [review]
You forgot `Use case ends.` for some use cases. Apart from that should be fine.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (Ma-Yueran) [reply docs/DeveloperGuide.md:387]: Is this the intended behavior of open command?

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Ma-Yueran) [reply docs/DeveloperGuide.md]: emmm, is to be invalid wrong here? I'm not sure.

Comment (luo-git) [reply docs/DeveloperGuide.md]: Yup it's a grammatical mistake

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (luo-git) [reply docs/DeveloperGuide.md:387]: Okay never mind.

**Review by luo-git — APPROVED** [review]
LGTM👍

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Ma-Yueran) [reply docs/DeveloperGuide.md]: Ok, changed

**Pull request merged**